src/clr-angular/data/datagrid/render/header-renderer.spec.ts

```diff
@@ -93,7 +99,190 @@ export default function(): void {
+      expect(column4InitialWidth).toBeGreaterThan(columnHeader4ResizerService.minColumnWidth);
+    });
+
+    it('expands other flexible headers if header width shrinks', function() {
+      const resizeBy = -20;
+
+      emulateResizeOnColumn(resizeBy, columnHeader1DraggableDirective);
+
+      expect(widthOf(columnHeader1Element)).toBe(column1InitialWidth + resizeBy);
+      expect(widthOf(columnHeader2Element)).toBeGreaterThan(column2InitialWidth);
+      expect(widthOf(columnHeader3Element)).toBe(
+        column3InitialWidth,
+        `A strict width shouldn't change when other header's width changes`
+      );
+      expect(widthOf(columnHeader4Element)).toBeGreaterThan(column4InitialWidth);
+    });
+
+    it('resized header should have fixed width class', function() {
+      expect(columnHeader1Element.classList.contains(STRICT_WIDTH_CLASS)).toBeFalse();
+      const resizeBy = -20;
+      emulateResizeOnColumn(resizeBy, columnHeader1DraggableDirective);
+      expect(columnHeader1Element.classList.contains(STRICT_WIDTH_CLASS)).toBeTrue();
+    });
+
+    it('shrinks other flexible headers if header width expands', function() {
+      const resizeBy = 20;
+
+      emulateResizeOnColumn(resizeBy, columnHeader1DraggableDirective);
+
+      expect(widthOf(columnHeader1Element)).toBe(column1InitialWidth + resizeBy);
+      expect(widthOf(columnHeader2Element)).toBeLessThan(column2InitialWidth);
+      expect(widthOf(columnHeader3Element)).toBe(
+        column3InitialWidth,
+        `A strict width shouldn't change when other header's width changes`
+      );
+      expect(widthOf(columnHeader4Element)).toBeLessThan(column4InitialWidth);
+    });
+
+    it("shouldn't shrink flexible headers below their min-width if header width expands by large amount", function() {
+      const resizeBy = 1000;
+      emulateResizeOnColumn(resizeBy, columnHeader1DraggableDirective);
+      expect(widthOf(columnHeader1Element)).toBe(column1InitialWidth + resizeBy);
+      expect(widthOf(columnHeader2Element)).toBe(120);
+      expect(widthOf(columnHeader3Element)).toBe(
+        200,
+        `A strict width shouldn't change when other header's width changes`
+      );
+      expect(widthOf(columnHeader4Element)).toBe(96);
+    });
+
+    it('gives header its min-width if a user tried to drag too much to left', function() {
+      const resizeBy = -1000;
+      emulateResizeOnColumn(resizeBy, columnHeader1DraggableDirective);
+      expect(widthOf(columnHeader1Element)).toBe(96);
+    });
+
+    it('emits new header width once resizing ends', function() {
+      expect(context.testComponent.newWidth).toBeUndefined();
+      const resizeBy = 20;
+      emulateResizeOnColumn(resizeBy, columnHeader3DraggableDirective);
+      expect(context.testComponent.newWidth).toBe(column3InitialWidth + resizeBy);
+    });
+  });
 }
 
 @Component({ template: `<clr-dg-column>Hello world</clr-dg-column>` })
```

Comment:
Minor request: Can we move these to the top of the file before the exported function? IIR we are going to organize test files like that moving forward.